thanos to aggregator migration doc (#865)
kubecost/docs: install-and-configure/install/multi-cluster/federated-etl/migrate-from-thanos-to-kubecost-2.0.md

# Migration from Thanos to Kubecost 2.0 (Aggregator)

> This tutorial is intended to help our users migrate from the legacy Thanos federation architecture to Kubecost 2.0 (Aggregator). There are a few requirements in order to successfully migrate to Kubecost 2.0. This new version of Kubecost includes a new backend (Aggregator) which handles the ETL data built from source metrics more efficiently. Kubecost 2.0 will provide new features,optimze the Kubecost primary UI and enhance the user experience.

Reviewer: ```suggestion
This tutorial is intended to help our users migrate from the legacy Thanos federation architecture to Kubecost 2.0 (Aggregator). There are a few requirements in order to successfully migrate to Kubecost 2.0. This new version of Kubecost includes a new backend (Aggregator) which handles the ETL data built from source metrics more efficiently. Kubecost 2.0 will provide new features, optimize the Kubecost primary UI and enhance the user experience.
```

## Key Changes
* Assets and Allocations are now paginated
   * FE/API users never have full set of information
   * Use offset/limit to govern pagination
* New data available for querying every 2 hours (can be adjusted)
* Embedded DuckDB database serves queries
   * Data no longer queried directly from bingen files
   * Substantial query speed improvements even when pagination not in effect
* Data ingested into independent component (Aggregator)

> * Idle (sharing), Cluster Management sharing, Network are computed a prior

Reviewer: ```suggestion
* Idle (sharing), Cluster Management sharing, and Network are computed a priori
```

* Distributed tracing integrated into core workflows
* No more pre-computed "agg stores"
   * Request-level caching still in effect

<details>

<summary>Diagram of Aggregator:</summary>

![aggregator-diagram](/images/aggregator/aggregator-diagram.png)

</details>

## Migration Path 

<details>

<summary>Diagram of Aggregator:</summary>

![migration-diagram](/images/aggregator/migration-diagram.png)

</details>

* All steps are done on the primary except for step 8.
* Nothing needs to be done on the secondary clusters for the migration to be successful. The Thanos sidecar on the secondary clusters will have no impact on the migration or functionality.
* Once Aggregator is enabled, all queries hit the Aggregator container and NOT cost-model via the reverse proxy.

> * ETL Utils does not destory the Thanos data, it creates additional directories in the object store.

Reviewer: ```suggestion
* ETL Utils does not destroy the Thanos data, it creates additional directories in the object store.
```
